crates/extensions/c8y_http_proxy/src/actor.rs

```diff
@@ -327,43 +427,30 @@ impl C8YHttpProxyActor {
         Ok(())
     }
 
-    async fn create_event_request(
-        &mut self,
-        event_type: String,
-        event_text: Option<String>,
-        event_time: Option<OffsetDateTime>,
-        child_device_id: Option<String>,
-    ) -> Result<C8yCreateEvent, C8YRestError> {
-        let device_internal_id = if let Some(device_id) = child_device_id {
-            self.get_c8y_internal_child_id(device_id).await?
-        } else {
-            self.end_point.get_c8y_internal_id().to_string()
-        };
-
-        let c8y_managed_object = C8yManagedObject {
-            id: device_internal_id,
-        };
-
-        Ok(C8yCreateEvent::new(
-            Some(c8y_managed_object),
-            event_type.clone(),
-            event_time.unwrap_or_else(OffsetDateTime::now_utc),
-            event_text.unwrap_or(event_type),
-            HashMap::new(),
-        ))
-    }
-
     async fn send_event_internal(
         &mut self,
-        c8y_event: C8yCreateEvent,
+        device_type: DeviceType,
+        create_event: impl Fn(String) -> C8yCreateEvent,
     ) -> Result<EventId, C8YRestError> {
-        let create_event_url = self.end_point.get_url_for_create_event();
+        // Get and set child device internal id
+        if !device_type.eq(&DeviceType::MainDevice) {
+            self.get_and_set_internal_id(device_type.clone()).await?;
+        }
```

Comment:
On a first look, it wasn't obvious to me as to why we're doing this here. But later understood.